packages/webapp/src/components/Animals/RemoveAnimalsModal/index.tsx

```diff
@@ -0,0 +1,185 @@
+};
+
+export default function RemoveAnimalsModal(props: RemoveAnimalsModalProps) {
+  const { register, handleSubmit, control, watch } = useForm<FormFields>();
+  const { t } = useTranslation();
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
+
+  const options = [
+    {
+      label: t('REMOVE_ANIMALS.SOLD'),
+      value: 'SOLD',
+    },
+    {
+      label: t('REMOVE_ANIMALS.SLAUGHTERED_FOR_SALE'),
+      value: 'SLAUGHTERED_FOR_SALE',
+    },
+    {
+      label: t('REMOVE_ANIMALS.SLAUGHTERED_FOR_CONSUMPTION'),
+      value: 'SLAUGHTERED_FOR_CONSUMPTION',
+    },
+    {
+      label: t('REMOVE_ANIMALS.NATURAL_DEATH'),
+      value: 'NATURAL_DEATH',
+    },
+    {
+      label: t('REMOVE_ANIMALS.CULLED'),
+      value: 'CULLED',
+    },
+    {
+      label: t('common:OTHER'),
+      value: 'OTHER',
+    },
+    {
+      label: t('REMOVE_ANIMALS.CREATED_IN_ERROR'),
+      value: 'CREATED_IN_ERROR',
+    },
+  ];
+
+  const selectedOption = watch(REASON);
+  const isCreatedInErrorSelected = selectedOption === options[options.length - 1].value;
+
+  // For styling dropdown options
+  const getReactSelectClassNames = (value: string) => {
+    if (!value) return '';
+    return clsx(
+      styles.dropDownOption,
+      value === options[options.length - 1].value
+        ? styles.dropDownOptionRed
+        : styles.dropDownOptionGreen,
+    );
+  };
+
+  return (
+    <>
+      {/* @ts-ignore */}
+      <Drawer
+        title={t('REMOVE_ANIMALS.REMOVE_ANIMALS')}
+        isOpen={props.isOpen}
+        onClose={props.onClose}
+      >
+        {props.showSuccessMessage ? (
+          <div className={styles.successMessage}>
+            <div>
+              <CheckIcon />
+            </div>
+            <p>{t('REMOVE_ANIMALS.REMOVED_AND_ARCHIVED')}</p>
+          </div>
+        ) : (
+          <form onSubmit={handleSubmit(props.onConfirm)}>
+            {isMobile ? (
+              <fieldset>
+                <legend>{t('REMOVE_ANIMALS.WHY')}</legend>
+                {options.map(({ label, value }) => (
+                  <label key={value} className={styles.mobileOption}>
+                    <input {...register(REASON)} value={value} type="radio" />
+                    <div
+                      className={clsx(
+                        styles.mobileOptionLabel,
+                        value === options[options.length - 1].value
+                          ? styles.mobileOptionLabelRed
+                          : styles.mobileOptionLabelGreen,
+                      )}
+                    >
+                      {label}
+                    </div>
+                  </label>
+                ))}
+              </fieldset>
+            ) : (
+              <Controller
+                name={REASON}
+                control={control}
+                render={({ field }) => (
+                  <ReactSelect
+                    {...field}
+                    // @ts-ignore
```

Comment:
@ts-ignore comment is a temporary fix to suppress the typescript errors for the component props. I'm not sure if there's an option in tsconfig that can work globally but it might be good to convert commonly used components like ReactSelect to typescript to avoid this.